<launch>
<arg name="sim" default="false" />
<group if="$(arg sim)">
  <!-- Rotate -90 degrees from tku robot to sim robot -->
  <node name="mapping" pkg="topic_tools" type="transform"
        args="$(env ROBOT_NS)/motion/cmd_vel /nubot1/nubotcontrol/velcmd nubot_common/VelCmd
              'nubot_common.msg.VelCmd(
               Vx=m.linear.y,
               Vy=m.linear.x * -1,
                w=m.angular.z)' --import nubot_common" />
  <node ns="$(env ROBOT_NS)" name="gazebo_relay" pkg="strategy" type="gazebo_relay.py" output="screen" />
</group>
<group ns="$(env ROBOT_NS)">
  <node name="core" pkg="strategy" type="core.py" args="$(arg sim)" output="screen" />
</group>
<rosparam command="load" file="$(find strategy)/cfg/topics.yaml" />
<!-- <node name="dynamic_reconfigure_load" pkg="dynamic_reconfigure" type="dynparam" args="load /core $(find strategy)/cfg/game_state.yaml" /> -->
<node name="mapping_ball_handle" pkg="topic_tools" type="transform"
      args="$(env ROBOT_NS)/strategy/state $(env ROBOT_NS)/motion/hold_ball std_msgs/Bool
            'std_msgs.msg.Bool(
             data=m.ball_is_handled)' --import std_msgs" />
</launch>
